overlord/auth/auth.go

```diff
@@ -69,14 +122,29 @@ func NewUser(st *state.State, username, email, macaroon string, discharges []str
 		return nil, err
 	}
 
+	if authStateData.MacaroonKey == nil {
+		authStateData.MacaroonKey, err = generateMacaroonKey()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	localMacaroon := macaroon
+	if username != "" {
```

Comment:
We don't want to make this conditional on username. Think of snapweb users.. these don't necessarily need to be associated with a local system user.